modules/ROOT/pages/values-and-types/property-structural-constructed.adoc

```diff
@@ -104,6 +105,56 @@ However, not all types can be used in all places.
 All Cypher types contain the `null` value. To make them not nullable, `NOT NULL` can be appended to the end of the type (e.g. `BOOLEAN NOT NULL`, `LIST<FLOAT NOT NULL>`).
 Note that closed dynamic types (`INNER_TYPE_1 | INNER_TYPE_2...`) cannot be appended with `NOT NULL`: all inner types must be nullable, or all appended with `NOT NULL`.
 
+== Type Normalization
+
+Cypher runs a normalization algorithm on all input types, simplifying the given type to a deterministic representation for equivalent types.
+Types are simplified to their default name (e.g. `BOOL` is simplified to `BOOLEAN`), encompassing types are absorbed (e.g. `LIST<BOOLEAN> | LIST<BOOLEAN | INTEGER>` is normalized to `LIST<BOOLEAN | INTEGER>`) and they are <<ordering-of-types, ordered>>.
+
+The type `PROPERTY VALUE` is expanded to a closed dynamic union of all valid property types, and if all types are represented, then the normalization would simplify to `ANY`.
+
+For example, given the closed dynamic type `BOOL | LIST<INT> | BOOLEAN | LIST<FLOAT | INT>`, the normalized type would be: `BOOLEAN | LIST<INTEGER | FLOAT>`.
+
+This normalization is run on types input in xref::syntax/expressions.adoc#type-predicate-expressions[type predicate expressions] and in xref::constraints/examples.adoc#constraints-examples-node-property-type[node] and xref::constraints/examples.adoc#constraints-examples-relationship-property-type[relationship] property type constraints.
```

Comment:
"input in" reads a bit strange, but maybe "used" is not correct? "given to", perhaps?

```suggestion
This normalization is run on types used in xref::syntax/expressions.adoc#type-predicate-expressions[type predicate expressions], and in xref::constraints/examples.adoc#constraints-examples-node-property-type[node] and xref::constraints/examples.adoc#constraints-examples-relationship-property-type[relationship] property type constraints.
```